native/chat/chat-thread-list.react.js

```diff
@@ -255,8 +259,17 @@ const styles = {
 };
 const stylesSelector = styleSelector(styles);
 
-export default connect((state: AppState) => ({
+export const HomeChatThreadList = connect((state: AppState) => ({
   chatListData: chatListData(state),
+  searchChatListData: chatBackgroundListData(state).concat(chatListData(state)),
```

Comment:
Doing this here means that `searchChatListData` will change every time the `connect` function is processed. The point of using selectors is memoization - meaning that the result will only change when the inputs change.

With this approach, any time anything in the Redux store changes (even unrelated), this component will have to be rerendered, and the result of `listDataSelector` will have to be recalculated. Making all the rest of the selectors useless.

The "obvious" solution here would be to create a new selector that is `chatBackgroundListData` and `chatListData` together. But I'm realizing perhaps my initial suggestion to simply replace the `chatListData` with a filtered one was bad.

Since we need the full `chatListData` for search, I think instead we should actually get rid of `chatBackgroundListData` and leave the `chatListData` as containing everything. Instead, the individual `ChatThreadList` can just have custom filter functions that filter the `chatListData` within the component to only show the relevant threads that match.

You could define `ChatThreadList` as a component that takes a function prop that can filter what it displays by default, and then memoizes the result of applying the function to the `chatListData` (as is currently done in `listDataSelector`).

`BackgroundChatThreadList` and `HomeChatThreadList` will then be defined as components that render a `ChatThreadList` with a custom filter function. You'll need to take care to make sure that filter function isn't recreated on every render. I'd suggest implementing `BackgroundChatThreadList` and `HomeChatThreadList` as hooks and using `useCallback` to define the filter function, but you could also define them as class components, with the filter functions defined as class properties.

Sorry for the wall of text - let me know if I can help explain over video chat!